File: tests/machine/x/dart/left_join_multi.dart
var customers = [
  {'id': 1, 'name': 'Alice'},
  {'id': 2, 'name': 'Bob'},
];

var orders = [
  {'id': 100, 'customerId': 1},
  {'id': 101, 'customerId': 2},
];

var items = [
  {'orderId': 100, 'sku': 'a'},
];

var result = (() {
  var _q0 = <dynamic>[];
  for (var o in orders) {
    for (var c in customers) {
      if (!(o['customerId'] == c['id'])) continue;
      var _jt1 = <dynamic>[];
      for (var i in items) {
        if (!(o['id'] == i['orderId'])) continue;
        _jt1.add(i);
      }
      if (_jt1.isEmpty) _jt1.add(null);
      for (var i in _jt1) {
        _q0.add({'orderId': o['id'], 'name': c['name'], 'item': i});
      }
    }
  }
  return _q0;
})();

void main() {
  print('--- Left Join Multi ---');
  for (var r in result) {
    print([r['orderId'], r['name'], r['item']].join(' '));
  }
}

File: tests/machine/x/cs/left_join_multi.cs
using System;
using System.Collections.Generic;
using System.Linq;

class Program {
    static void Main() {
        var customers = new List<dynamic> { new Dictionary<dynamic, dynamic> { { "id", 1 }, { "name", "Alice" } }, new Dictionary<dynamic, dynamic> { { "id", 2 }, { "name", "Bob" } } };
        var orders = new List<dynamic> { new Dictionary<dynamic, dynamic> { { "id", 100 }, { "customerId", 1 } }, new Dictionary<dynamic, dynamic> { { "id", 101 }, { "customerId", 2 } } };
        var items = new List<dynamic> { new Dictionary<dynamic, dynamic> { { "orderId", 100 }, { "sku", "a" } } };
        var result = new Func<List<dynamic>>(() => {
    var _res = new List<dynamic>();
    foreach (var o in orders) {
        foreach (var c in customers) {
            if (!((o.customerId == c.id))) continue;
            foreach (var i in items) {
                if (!((o.id == i.orderId))) continue;
                _res.Add(new Dictionary<dynamic, dynamic> { { "orderId", o.id }, { "name", c.name }, { "item", i } });
            }
        }
    }
    return _res;
})();
        Console.WriteLine("--- Left Join Multi ---");
        foreach (var r in result) {
            Console.WriteLine(string.Join(" ", new [] { Convert.ToString(r.orderId), Convert.ToString(r.name), Convert.ToString(r.item) }));
        }
    }
}
